pkg/prometheus/promcfg.go

```diff
@@ -2575,6 +2576,65 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// EC2SDConfig
+	if len(sc.Spec.EC2SDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.EC2SDConfigs))
+		for i, config := range sc.Spec.EC2SDConfigs {
+			configs[i] = []yaml.MapItem{
+				{
+					Key:   "region",
+					Value: config.Region,
```

Comment:
I think that we need to check for nil.

Reply:
per prometheus region can be empty but in that case other fields are expected to get instance metadata. So I had this thought https://github.com/prometheus-operator/prometheus-operator/pull/5902#discussion_r1320312099 whether to make this required field till we implement/plan to support other methods other than with api access keys